pkg/list/values/extract.go

```
@@ -0,0 +1,249 @@
+	if len(result) == 0 {
+		return nil, &errors.QueryError{
+			Query: query,
+			Cause: &errors.NoValuesFoundError{Component: "query", Query: query},
+		}
+	}
+
+	return result, nil
+}
+
+// getValueFromPath gets a value from a nested map using a dot-separated path.
+func getValueFromPath(data map[string]interface{}, path string) interface{} {
```

golangci-lint (1.62.2)

cognitive complexity 47 of func `getValueFromPath` is high (> 20)

(gocognit)